### ISSUE TITLE
Dawn backends: Compilation failure due to missing Implementation IR

### ISSUE THREAD

**Issue by stubbiali** [GridTools/gt4py#144]
I cannot compile any stencil with Dawn because of this error:
```
File "/project/s299/subbiali/tasmania-master/docker/external/gt4py/src/gt4py/gtscript.py", line 208, in _decorator
    externals=externals or {},
File "/project/s299/subbiali/tasmania-master/docker/external/gt4py/src/gt4py/loader.py", line 66, in gtscript_loader
    stencil_class = load_stencil("gtscript", backend, definition_func, externals, build_options)
File "/project/s299/subbiali/tasmania-master/docker/external/gt4py/src/gt4py/loader.py", line 55, in load_stencil
    stencil_class = backend.generate(stencil_id, definition_ir, definition_func, build_options)
File "/project/s299/subbiali/tasmania-master/docker/external/gt4py/src/gt4py/backend/dawn_backends.py", line 502, in generate
    stencil_id, definition_ir, options, module_info=module_info, **kwargs
File "/project/s299/subbiali/tasmania-master/docker/external/gt4py/src/gt4py/backend/base.py", line 875, in __call__
    return super().__call__(stencil_id, definition_ir, options, module_info, **kwargs)
File "/project/s299/subbiali/tasmania-master/docker/external/gt4py/src/gt4py/backend/base.py", line 680, in __call__
    imports = self.generate_imports()
File "/project/s299/subbiali/tasmania-master/docker/external/gt4py/src/gt4py/backend/base.py", line 881, in generate_imports
    if self.implementation_ir.multi_stages:
AttributeError: 'NoneType' object has no attribute 'multi_stages'
```
I may guess the error lies in the fact that Dawn-based backends do not generate any Implementation IR, so the attribute `implementation_ir` of `BaseModuleGenerator` is never set. Moving [these two lines](https://github.com/GridTools/gt4py/blob/e9fe299319c8d12a28bbc45135da76c427d7fa23/src/gt4py/backend/base.py#L672-L673) out of the if-statement does the trick, but I'm sure there's a cleaner and more elegant solution.

**Comment by egparedes** [comment]
This should be fixed soon by #134 and  #146

**Comment by stubbiali** [comment]
Dawn backends compile fine now. I will close the issue.
